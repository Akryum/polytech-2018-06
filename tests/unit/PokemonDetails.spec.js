import PokemonDetails from '@/components/PokemonDetails.vue'
import { mount } from '@vue/test-utils'

const pokemon = {
  id: '46',
  name: 'paras',
  __typename: 'Pokemon',
  hp: 35,
  attack: 70,
  defense: 55,
  types: [
    { id: 'bug', __typename: 'PokemonType' },
    { id: 'grass', __typename: 'PokemonType' }
  ]
}

describe('PokemonDetails', () => {
  const mocks = {
    $route: { params: { id: 46 } },
    $apollo: { loading: false }
  }

  const stubs = {
    'BaseImage': '<div data-stub-base-image v-bind="$attrs"></div>'
  }

  test('snapshot', () => {
    const wrapper = mount(PokemonDetails, {
      propsData: {
        id: 46
      },
      data: () => ({
        pokemon
      }),
      mocks,
      stubs
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.name').text()).toBe('paras')
  })

  test('snapshot: pokemon not found', () => {
    const wrapper = mount(PokemonDetails, {
      propsData: {
        id: 'foo'
      },
      data: () => ({
        pokemon: null
      }),
      mocks,
      stubs
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.empty').text()).toBe('Pokemon not found')
  })

  test('event: toggle-favorite', () => {
    const wrapper = mount(PokemonDetails, {
      propsData: {
        id: 46
      },
      data: () => ({
        pokemon
      }),
      mocks,
      stubs
    })
    wrapper.find('.toggle-favorite').trigger('click')
    // L'événement a été appelé une fois
    expect(wrapper.emitted('toggle-favorite')).toBeTruthy()
    // L'argument en paramètre était 2
    expect(wrapper.emitted('toggle-favorite')[0]).toEqual([46])
  })
})
