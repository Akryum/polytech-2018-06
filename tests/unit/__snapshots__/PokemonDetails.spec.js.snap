// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetails snapshot 1`] = `
<div class="pokemon-details">
  <div class="id"><span class="symbol">#</span> <span class="number">46</span></div>
  <div class="name">paras</div>
  <div class="types">
    <div class="type">bug</div>
    <div class="type">grass</div>
  </div>
  <div data-stub-base-image="" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/46.png" alt="Pokemon 46" class="preview"></div>
  <button class="toggle-favorite">
    Toggle favorite
  </button>
</div>
`;

exports[`PokemonDetails snapshot: pokemon not found 1`] = `
<div class="pokemon-details">
  <div class="empty">Pokemon not found</div>
</div>
`;
